### PR TITLE
Turn on all Kubernetes CWL tests that are expected to work on Singularity

### LINKED ISSUE #3542: Fix CWL 1.2 conformance test 85 to reliably succeed on Kubernetes
Sometimes this test fails when run on Kubernetes. It should always pass, and it should be made a non-optional part of CI.

┆Issue is synchronized with this [Jira Task](https://ucsc-cgl.atlassian.net/browse/TOIL-855)
┆Epic: CWL v1.2 support
┆Issue Number: TOIL-855

### PR DESCRIPTION
**Pull request by adamnovak**
This is an alternative to #3719 and will fix #3542 through #3564 if it passes the CWL 1.2 Kubernetes conformance tests consistently. 

## Changelog Entry
To be copied to the [draft changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog) by merger:

 * Kubernetes batch system now tested for as much conformance to CWL 1.2 as is possible with Singularity containers

## Reviewer Checklist

 * [ ] Make sure it is coming from `issues/XXXX-fix-the-thing` in the Toil repo, or from an external repo.
    * [ ] If it is coming from an external repo, make sure to pull it in for CI with:
        ```
        contrib/admin/test-pr otheruser theirbranchname issues/XXXX-fix-the-thing
        ```
    * [ ] If there is no associated issue, [create one](https://github.com/DataBiosphere/toil/issues/new).
* [ ] Read through the code changes. Make sure that it doesn't have:
    * [ ] Addition of trailing whitespace.
    * [ ] New variable or member names in `camelCase` that want to be in `snake_case`.
    * [ ] New functions without [type hints](https://docs.python.org/3/library/typing.html).
    * [ ] New functions or classes without informative docstrings.
    * [ ] Changes to semantics not reflected in the relevant docstrings.
    * [ ] New or changed command line options for Toil workflows that are not reflected in `docs/running/cliOptions.rst`
    * [ ] New features without tests.
* [ ] Comment on the lines of code where problems exist with a review comment. You can shift-click the line numbers in the diff to select multiple lines.
* [ ] Finish the review with an overall description of your opinion.

## Merger Checklist

* [ ] Make sure the PR passes tests.
* [ ] Make sure the PR has been reviewed **since its last modification**. If not, review it.
* [ ] Merge with the Github "Squash and merge" feature.
    * [ ] If there are multiple authors' commits, add [Co-authored-by](https://github.blog/2018-01-29-commit-together-with-co-authors/) to give credit to all contributing authors.
* [ ] Copy its recommended changelog entry to the [Draft Changelog](https://github.com/DataBiosphere/toil/wiki/Draft-Changelog).
* [ ] Append the issue number in parentheses to the changelog entry.

### CODE REVIEW

**Comment by adamnovak** [comment]
@mr-c the only failing test here is 175:

```
Traceback (most recent call last):
  File "/builds/databiosphere/toil/venv/bin/toil-cwl-runner", line 33, in <module>
    sys.exit(load_entry_point('toil', 'console_scripts', 'toil-cwl-runner')())
  File "/builds/databiosphere/toil/src/toil/cwl/cwltoil.py", line 3151, in main
    outobj = toil.start(wf1)
  File "/builds/databiosphere/toil/src/toil/common.py", line 841, in start
    return self._runMainLoop(rootJobDescription)
  File "/builds/databiosphere/toil/src/toil/common.py", line 1127, in _runMainLoop
    return Leader(config=self.config,
  File "/builds/databiosphere/toil/src/toil/leader.py", line 278, in run
    raise FailedJobsException(self.config.jobStore, self.toilState.totalFailedJobs,
toil.leader.FailedJobsException: The job store 'aws:us-west-2:toil-cwl-a76b122c-beed-4c88-b4a8-b9a7e32e32f4' contains 1 failed jobs: 'CWLJob' -c echo 'moo' > cow 12e00431-af52-438d-88fd-03f161515cf5 v6
Log from job "'CWLJob' -c echo 'moo' > cow 12e00431-af52-438d-88fd-03f161515cf5 v6" follows:
=========>
	/.singularity.d/actions/exec: exec: line 9: illegal option -c
	[2021-07-20T17:11:56+0000] [MainThread] [E] [cwltool] [job docker-run-cmd.cwl] Job error:
	("Error collecting output for parameter 'cow':\ntests/docker-run-cmd.cwl:11:7: Did not find output file with glob pattern: '['cow']'", {})
	Traceback (most recent call last):
	  File "/usr/local/lib/python3.8/dist-packages/toil/worker.py", line 402, in workerScript
	    job._runner(jobGraph=None, jobStore=jobStore, fileStore=fileStore, defer=defer)
	  File "/usr/local/lib/python3.8/dist-packages/toil/job.py", line 2382, in _runner
	    returnValues = self._run(jobGraph=None, fileStore=fileStore)
	  File "/usr/local/lib/python3.8/dist-packages/toil/job.py", line 2303, in _run
	    return self.run(fileStore)
	  File "/usr/local/lib/python3.8/dist-packages/toil/cwl/cwltoil.py", line 1872, in run
	    raise cwltool.errors.WorkflowException(status)
	cwltool.errors.WorkflowException: permanentFail
	[2021-07-20T17:11:56+0000] [MainThread] [E] [toil.worker] Exiting the worker because of a failed job on host toiltest-toil-d6e9d4fe-5445-4c32-9d9b-0f1aef5d265c-1-xjnvv
<=========
Test 175 failed: toil-cwl-runner --disableCaching=True --clean=always --logDebug --setEnv=SINGULARITY_DOCKER_HUB_MIRROR --batchSystem=kubernetes --outdir=/tmp/tmpk6eaekt0 --quiet tests/docker-run-cmd.cwl tests/empty.json
Test Docker ENTRYPOINT usage
Returned non-zero
None
```

That runs this workflow:
```
class: CommandLineTool
cwlVersion: v1.2
requirements:
  DockerRequirement:
    dockerPull: bash:4.4.12
inputs: []
outputs:
  cow:
    type: File
    outputBinding:
      glob: cow
baseCommand: ["-c", "echo 'moo' > cow"]
```

I think it isn't passing because Singularity isn't being invoked right by `cwltool`. Is `cwltool` expected to pass all the conformance tests, and in particular this one, with `--singularity`?

**Comment by mr-c** [comment]
@adamnovak 

Here are the conformance tests we skip when in singularity mode: https://github.com/common-workflow-language/cwltool/blob/7094ede917c2d5b16d11f9231fe0c05260b51be6/conformance-test.sh#L99

Including this test, yes. Due to https://github.com/hpcng/singularity/issues/3673

**Comment by adamnovak** [comment]
OK, I pulled in a `cwltest` new enough to support skipping tests, and set the Kubernetes tests to skip the same tests as cwltool does for singularity (except for one that is supposed to be fixed now).

File: setup.py [hunk 1]
```diff
@@ -68,6 +69,7 @@ def run_setup():
         pycryptodome]
     cwl_reqs = [
         cwltool,
+        cwltest,
```
Comment (mr-c) [inline setup.py:72]: Please don't; regular `toil-cwl-runner` users don't need cwltest. How about as another extra named `cwl-dev`?

**Review by mr-c — CHANGES_REQUESTED** [review]

File: setup.py (hunk 1 above)
Comment (DailyDreaming) [reply setup.py:72]: @mr-c +1.  Though all dev reqs as `dev` might be good too (maybe getting rid of `make prepare`?).

Comment (adamnovak) [reply setup.py:72]: Oh, we already have `requirements-dev.txt` which is how we get it in the first place. I thought `cwltool` was bringing it along. I will use that mechanism instead.

**Review by mr-c — APPROVED** [review]

**Pull request merged**